rl4co/envs/common/distribution_utils.py

```diff
@@ -0,0 +1,184 @@
+import torch
+
+class Cluster():
```

Comment:
I like this!  @cbhua I think we should train some model with say TSP50 / CVRP50 with a mixed distribution and test its generalization performance

Minor comment: Shouldn't this be a subclass of `torch.distributions.distribution.Distribution`?

Reply:
Yes actually this class is what we are missing in the distribution! This could be used by various environments' generator. 

About the experiment, we want to test the distribution generalization ability right?